src/jdk.compiler/share/classes/com/sun/tools/javac/tree/JCTree.java

```diff
@@ -683,8 +701,35 @@ protected JCImport(JCFieldAccess qualid, boolean importStatic) {
         @DefinedBy(Api.COMPILER_TREE)
         public boolean isStatic() { return staticImport; }
         @DefinedBy(Api.COMPILER_TREE)
+        public boolean isModule() { return false; }
+        @DefinedBy(Api.COMPILER_TREE)
         public JCFieldAccess getQualifiedIdentifier() { return qualid; }
 
+        @Override
+        public Tag getTag() {
+            return IMPORT;
+        }
+    }
+
+    /**
+     * A module import clause.
+     */
+    public static class JCModuleImport extends JCImportBase {
+        /** The module name. */
+        public JCExpression module;
```

Comment:
Does it need to be an expression? Or is a name enough?

Reply:
Expression is more consistent with e.g. the module directives (`JCRequires` has `JCExpression moduleName`). Also allows us to keep positions for the name.